app/state_machines/form_state_machine.rb

```diff
@@ -0,0 +1,29 @@
+module FormStateMachine
+  extend ActiveSupport::Concern
+
+  included do
+    include AASM
+
+    enum :state, {
+      draft: "draft",
+      live: "live",
+    }
+
+    aasm column: :state, enum: true do
+      state :draft, initial: true
+      state :live
+
+      event :make_form_live do
+        transitions from: :draft, to: :live, guard: Proc.new { task_status_service.mandatory_tasks_completed? }
+      end
+
+      event :make_changes_live do
+        transitions from: :live, to: :draft, guard: Proc.new { has_live_version }
```

Comment:
I'd have expected `:make_changes_live` to transition `from: :draft, to: :live` but its currently the other way around?

Reply:
I was trying to reuse the same language we have in the ui but yes you are correct in this instance it should actually be "draft" to "live" and there should be another event that gets triggered when anything on the form/pages/routing conditions changes.

I think having clear language between the states is what we need.

Another way is to think of "Live" forms as just "versions" of a Form. the latest version of the finalised form are published on govuk. the versioning is just a means for us to "make a commit"  of a form in its current state. This way our "Form" model works and "MadeLiveVersions" are snapshots of the form.

Reply:
resolving this conversation because the event/state names have changed a lot since this conversation took place.